iquip/apps/scheduler.py

```diff
@@ -244,10 +258,68 @@ def paint(self,
     def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:  # pylint: disable=unused-argument
         """Overridden."""
         data = index.data(Qt.DisplayRole)
+        if data is None:
+            return QSize(0, 0)
         experimentView = ExperimentView(data)
         return experimentView.sizeHint()
 
 
+class _ExperimentQueueFetcherThread(QThread):
+    """QThread for fetching the queued experiments from the proxy server.
+
+    Signals:
+        fetched(experimentList, runningExperiment):
+          The experiment queue and currently running experiment are fetched.
+
+    Attributes:
+        ctrl: A dictionary that controls the thread from outside.
+    """
+
+    fetched = pyqtSignal(list, object)
+
+    def __init__(
+        self,
+        callback: Callable[[SubmittedExperimentInfo, None], List[SubmittedExperimentInfo]],
+        parent: Optional[QObject] = None
+    ):
+        """Extended.
+
+        Args:
+            callback: The callback method called after this thread is finished.
+        """
+        super().__init__(parent=parent)
+        self.fetched.connect(callback, type=Qt.QueuedConnection)
+        self.ctrl = {"break": False}
```

Comment:
It is just a question!
How come you define `ctrl` as a dictionary, not a boolean variable?
Is it for scalability?

Reply:
Also, the unittest still failed, right?

Reply:
> How come you define `ctrl` as a dictionary, not a boolean variable?

It is from the site you linked, and I guess it has to be mutable to make the value affect each other.

Reply:
> Also, the unittest still failed, right?

Yes...

Reply:
> It is from the site you linked, and I guess it has to be mutable to make the value affect each other.

Umm.. I think setting `ctrl` as a boolean value is okay and could you check this?

Reply:
> Yes...

How about waiting the thread at the end of each unittest, as follows?
```python
app.thread.ctrl["break"] = True  # app.thread.isRunning = True
app.thread.wait()
```